### DISCUSSION TITLE
How can i contribute to wrong routes?

### DISCUSSION THREAD

**Discussion by sami155** [organicmaps/organicmaps#486] [Q&A] [answered]
I compared some routes via Bicycle with G-Maps (evil) and Organic maps.
And many times both Maps apps give me unnecessary long paths to my destination.
In Google maps i can just shake my phone and then I give my feedback to the Google team.

But how can I give you guys feedback to unnecessary long routes? And how does it work? Is there a algorithm that can be fed with data?

I would love to help you.
See ya

**Comment by rtsisyk** [comment]
> But how can I give you guys feedback

Absolutely! Please feel free to file a [ticket](https://github.com/organicmaps/organicmaps/issues) which examples of problematic routes.

**Comment by biodranik** [comment, marked as answer]
Generally, our routes are built from OpenStreetMap data, so if you improve data there directly, the routes will improve in Organic Maps too. There may be some bugs in our code though, and map updates are not delivered instantly (yet), so it makes sense to check routes here first: http://map.project-osrm.org/
